src/polyfem/io/OutData.cpp

```diff
@@ -48,6 +49,74 @@ extern "C" size_t getPeakRSS();
 
 namespace polyfem::io
 {
+	namespace
+	{
+		void compute_traction_forces(const State &state, const Eigen::MatrixXd &solution, Eigen::MatrixXd &traction_forces, bool skip_dirichlet = true)
+		{
+
+			int actual_dim = 1;
+			if (!state.problem->is_scalar())
+				actual_dim = state.mesh->dimension();
+			else
+				return;
+
+			const std::vector<basis::ElementBases> &bases = state.bases;
+			const std::vector<basis::ElementBases> &gbases = state.geom_bases();
+
+			Eigen::MatrixXd uv, samples, gtmp, rhs_fun, deform_mat, trafo;
+			Eigen::VectorXi global_primitive_ids;
+			Eigen::MatrixXd points, normals;
+			Eigen::VectorXd weights;
+			polyfem::assembler::ElementAssemblyValues vals;
+
+			traction_forces.setZero(state.n_bases * actual_dim, 1);
+
+			for (const auto &lb : state.total_local_boundary)
+			{
+				const int e = lb.element_id();
+				bool has_samples = utils::BoundarySampler::boundary_quadrature(lb, state.n_boundary_samples(), *state.mesh, false, uv, points, normals, weights, global_primitive_ids);
+
+				if (!has_samples)
+					continue;
+
+				const basis::ElementBases &gbs = gbases[e];
+				const basis::ElementBases &bs = bases[e];
+
+				vals.compute(e, state.mesh->is_volume(), points, bs, gbs);
+
+				for (int n = 0; n < normals.rows(); ++n)
+				{
+					normals.row(n) = normals.row(n) * vals.jac_it[n];
```

Comment:
now this is the rest config, is this correct?

Reply:
Correct. Should I add this to the name?